pymatgen/transformations/advanced_transformations.py

```diff
@@ -1360,3 +1361,213 @@ def inverse(self):
+            scs_per_displacement_val (int): number of perturbed supercells to generate for each unique
+                displacement value.
+
+            floor_displacement (Optional float): If None (default), then for a given perturbed supercell, all atoms will
+                move the same distance from their original locations. If float, then for a given perturbed supercell,
+                the distances that atoms move will be uniformly distributed from a minimum value of 'floor_displacement'
+                to the displacement value 'displacement_val'.
+        Returns:
+            List of randomly displaced supercell structures (List of Structures)
+    """
+
+    def __init__(self, original_supercell,
+                 max_displacement_val,
+                 min_displacement_val,
+                 num_displacements,
+                 scs_per_displacement_val,
+                 floor_displacement=None):
+
+        self.original_supercell = original_supercell
+        self.max_disp_val = max_displacement_val
+        self.min_disp_val = min_displacement_val
+        self.num_disps = num_displacements
+        self.scs_per_disp_val = scs_per_displacement_val
+
+        if floor_displacement is not None:
+            self.floor_disp = float(floor_displacement)
+        else:
+            self.floor_disp = None
+
+        self.disp_vals = np.linspace(min_displacement_val, max_displacement_val, num=num_displacements)
+        self.perturbed_supercells = self.generate_transformations()
+
+    def random_displacements(self, natom, rmax, rmin, dim=3):
+        #*** Adapted from csld.util.mathtool
+
+        # Generate matrix of size (natom, 3) where each row is a Gaussian random vector
+        # If 'rmax'=None, then all vectors will have magnitude 'rmin'.
+        # Else, magnitudes are uniformly distributed between 'rmin' and 'rmax'.
+
+        dx = np.random.normal(size=(natom, dim)) #Gaussian sampled displacements. Matrix size: (natom, 3)
+        dx_norms = np.linalg.norm(dx, axis=1)
+        veclen = np.full(natom, rmax) if rmin is None else np.random.uniform(rmin, rmax, natom)
+
+        if not 0 in dx_norms:
+            return dx * (veclen / dx_norms)[:, None]
+        else:
+            self.random_displacements(natom, rmax, rmin, dim)
+
+
+    def perturb_supercell(self, structure, max_disp, floor_disp):
+        # *** Adapted from CSLD's 'polaron_main' file
+
+        na = structure.num_sites
+        max_disp = float(max_disp)
+        if isinstance(max_disp, float):
+            dr = self.random_displacements(na, max_disp, floor_disp) #generate random displacements
+
+            # Perturb structure with the random displacements
+            for i in range(len(structure._sites)):
+                structure.translate_sites([i], dr[i], frac_coords=False, to_unit_cell=True)
+            return structure
+        else:
+            print('Displacement entered is not a float.')
+
+    def generate_transformations(self):
```

Comment:
What is this for? 

I think this should instead be refactored to override `apply_transformation` superfunction